bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/OpenhabGraalJSScriptEngine.java

```diff
@@ -304,13 +311,17 @@ protected void beforeInvocation() {
 
         initialized = true;
 
+        logger.debug(
+                "Engine '{}': isScriptFile(): {}, isScriptAction(): {}, isScriptCondition(): {}, isTransformation(): {}",
+                engineIdentifier, isScriptFile(), isScriptAction(), isScriptCondition(), isTransformation());
+
         try {
             logger.debug("Evaluating cached global script for engine '{}' ...", engineIdentifier);
             delegate.getPolyglotContext().eval(GLOBAL_SOURCE);
 
             if (configuration.isInjectionEnabledForAllScripts()
-                    || (isUiBasedScript() && configuration.isInjectionEnabledForUiBasedScript())
-                    || (isTransformationScript() && configuration.isInjectionEnabledForTransformations())) {
+                    || (!isScriptFile() && configuration.isInjectionEnabledForUiBasedScript())
```

Comment:
Would it be possible to rename these methods too?

Reply:
Yes, it should :+1: